integration-tests/tests/src/tests/class-models.ts

```diff
@@ -75,8 +75,8 @@ describe("Class models", () => {
   });
 
   describe("#constructor", () => {
-    type UnmanagedPerson = Partial<Person> & Pick<Person, "name">;
-    class Person extends Realm.Object<UnmanagedPerson> {
+    // The Pick and Partial is needed to correctly reflect the defaults
+    class Person extends Realm.Object<Pick<Person, "name"> & Partial<Person>> {
```

Comment:
Is this since "id", "age" and "friends" default to something?  Will all users of class based models have to type this syntax themselves?  Perhaps we could build a utility type for this to make it more convenient.

Reply:
With the use of property initialisers to declare defaults (as proposed by TypeScript-first Models), I believe this issue will go away.

Reply:
Then I am happy 🙂